source/extensions/transport_sockets/http_11_proxy/connect.h

```diff
@@ -81,6 +81,8 @@ class SelfContainedParser : public Http::Http1::ParserCallbacks {
   }
   Http::Http1::CallbackResult onHeadersComplete() override {
     headers_complete_ = true;
+    status_200_ = parser().getStatus() != Http::Http1::ParserStatus::Error &&
+                  parser().statusCode() == Http::Code::OK;
```

Comment:
Why not put this operation into the `status200()` function rather than precalculating and storing? (If there is a reason, a comment explaining that reason would be good.)

Reply:
(I see the reason in #34096 now, so explanatory comment is the request!)